src/Storages/ObjectStorage/DataLakes/Common.cpp

```diff
@@ -11,7 +11,7 @@ std::vector<String> listFiles(
     const StorageObjectStorage::Configuration & configuration,
     const String & prefix, const String & suffix)
 {
-    auto key = std::filesystem::path(configuration.getPath()) / prefix;
```

Comment:
Why to use filename instead of the whole path?

Reply:
Understand. You changed the Path definition in StorageObjectStorage::Configuration. How about changing it to be PathInfo.

Reply:
And the attribute of `filename` is also a bit confusing. Maybe it's better to change it to be path.